spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Span.java

```diff
@@ -41,18 +41,17 @@
 	 * The spanId is immutable and cannot be changed. It is safe to access this from
 	 * multiple threads.
 	 */
-	String getSpanId();
+	Long getSpanId();
 
 	/**
 	 * A pseudo-unique (random) number assigned to the trace associated with this span
 	 */
-	String getTraceId();
+	long getTraceId();
 
 	/**
 	 * Return a unique id for the process from which this Span originated.
 	 * <p/>
 	 * <p/>
-	 * Will never be null.
```

Comment:
hah!! document when it can be null, or TODO if not yet known. The null-ability in various fields has been a bug factory. I'd also encourage us to make an internal Nullable annotation to re-inforce the point.